src/documents/DataSourceAPI.md

```diff
@@ -8,9 +8,114 @@ database, cookie, in memory storage and etc). The specifics for accessing the da
 implementation. Additionally the payload of a request may be configured to be processed through an encoding/decoding
 step and other transformations during the get and set workflows.
 
-## Grades
+Concrete DataSource implementations include `fluid.dataSource.url` which read and write data over HTTP both in the
+browser and in node.js, as well as `kettle.dataSource.file` available in node.js acting on the filesystem, and
+`fluid.prefs.cookieStore` which acts on cookies in the browser. Mixin grades include `kettle.dataSource.CouchDB` which
+is capable of imbuing a core DataSource with CouchDB-like persistence semantics.
```

Comment:
It doesn't appear that these concrete grades, with the exception of `fluid.prefs.cookieStore`, are actually included in Infusion 3. I suppose these may exists in other versions or applications, but it's a bit confusing here as it seems to imply that they are available for use with the current version of Infusion.

Reply:
All the other ones are supplied in kettle which works with Infusion 3.x. Would be ok to just put a parenthetical comment after fluid.dataSource.url saying (available in Infusion 4.x)?

Reply:
Maybe link out to the kettle for the kettle ones, or mention that somewhere for those unfamiliar with it. Adding "(available in Infusion 4.x) would be fine for the fluid.dataSource.url, because we're planning a quick turn around for that version.